build_tools/python/e2e_test_framework/models/matmul.py

```diff
@@ -0,0 +1,69 @@
+## Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Defines Matmul microbenchmarks."""
+
+from e2e_test_framework import unique_ids
+from e2e_test_framework.definitions import common_definitions
+
+MATMUL_3456X1024X2048_FP16_MLIR = common_definitions.Model(
+    id=unique_ids.UB_MATMUL_3456X1024X2048_FP16_MLIR,
+    name="matmul_3456x1024x2048_f16t_f16t_f16t_tile_config_default",
```

Comment:
Does it make sense to only have one f16 in the name? Same to other microbenchmarks

Reply:
Technically all 3 could be different. I would be ok with abbreviating for the special case where they're all the same though. Depends on what's easier for the people looking at these benchmarks to read

Reply:
Let's try and keep the "name" short. Abbreviated to f16 or f32.

Reply:
Yes, all three could be different datatypes and also layout. If we have to support all the cases supported by libraries, we need to have various combination of datatypes and layout running in our CI. Thus, I would champion adopting a descriptive naming convention to cover all the possibilities. 

`f16t_f16t_f16t` is of the form `{dataTypeA}{layoutA}_{dataTypeB}{layoutB}_{dataTypeC}{layoutC}`

`t` here means row-major and `n` would mean column-major as per cuBLAS matmul convention. It is ok to abbreviate for now as we only support: (F16 Tensor Core)`f16 <= f16 * f16 + f16` and (F32 Tensor Core)`f32 <= f32* f32 + f32`. However, as we add more data types and layout we might have expand these names in the future.